### PR TITLE
DOCSP-45076: Document that count supports different filters from countDocuments

### PR DESCRIPTION
**Pull request by mcmorisi**
# Pull Request Info

[PR Reviewing Guidelines](https://github.com/mongodb/docs-golang/blob/master/REVIEWING.md)

JIRA - <https://jira.mongodb.org/browse/DOCSP-45076>

### Staging Links

<li><a href=https://deploy-preview-93--docs-ruby.netlify.app/reference/aggregation>reference/aggregation</a></li>

## Self-Review Checklist

- [x] Is this free of any warnings or errors in the RST?
- [x] Did you run a spell-check?
- [x] Did you run a grammar-check?
- [x] Are all the links working?
- [x] Are the [facets and meta keywords](https://wiki.corp.mongodb.com/display/DE/Docs+Taxonomy) accurate?

### CODE REVIEW

File: source/reference/aggregation.txt [hunk 1]
```diff
@@ -98,6 +98,14 @@ find the total number of documents which have the exact array
 
   count = coll.count({ 'categories' => [ 'Chinese', 'Seafood' ] })
 
+.. note::
+
+   The ``count`` method is not the same as the ``countDocuments``
+   method and may not support the same filters. ``countDocuments`` uses the
+   ``$match`` stage of the aggregation pipeline to filter documents. To learn more
+   about limitations of the ``$match`` stage, see :manual:`$match Restrictions </reference/operator/aggregation/match/#restrictions>`
+   in the MongoDB Server documentation.
```
Comment (mcmorisi) [inline source/reference/aggregation.txt:107]: FYI I can't add a source constant to the snooty file without breaking the build – unsure if it's a quirk fo the Ruby driver (I notice that it doesn't have many constants in the snooty file as is).

Comment (norareidy) [inline source/reference/aggregation.txt:107]: S:
```suggestion
   in the MongoDB Server manual.
```

Comment (norareidy) [reply source/reference/aggregation.txt:107]: hm not sure why that is. The {+mdb-server+} constant was added in the standardization branch, but I think it's fine to not use the constant here

**Review by norareidy — APPROVED** [review]
LGTM with a small suggestion

**Pull request merged**